tests/monitor.cpp

```diff
@@ -150,3 +150,89 @@ TEST_CASE("monitor from move assigned socket", "[monitor]")
     // failing
 }
 #endif
+
+#if defined(ZMQ_BUILD_DRAFT_API) && defined(ZMQ_CPP11)                              \
+  && !defined(ZMQ_CPP11_PARTIAL) && defined(ZMQ_HAVE_POLLER)
+#include "zmq_addon.hpp"
+
+TEST_CASE("poll monitor events using active poller", "[monitor]")
+{
+    // define necessary class for test
+    class test_monitor : public zmq::monitor_t
+    {
+      public:
+        void init(zmq::socket_t &socket,
+                  const char *const addr_,
+                  int events = ZMQ_EVENT_ALL)
+        {
+            zmq::monitor_t::init(socket, addr_, events);
+        }
+
+        void addToPoller(zmq::active_poller_t &inActivePoller)
+        {
+            inActivePoller.add(
+              monitor_socket(), zmq::event_flags::pollin,
+              [&](zmq::event_flags ef) { process_event(static_cast<short>(ef)); });
+        }
+
+        void on_event_accepted(const zmq_event_t &event_,
+                                       const char *addr_) override
+        {
+            clientAccepted++;
+        }
+        void on_event_disconnected(const zmq_event_t &event,
+                                   const char *const addr) override
+        {
+            clientDisconnected++;
+        }
+
+        int clientAccepted = 0;
+        int clientDisconnected = 0;
+    };
+
+    //Arrange
+    int messageCounter = 0;
+    const char monitorAddress[] = "inproc://monitor-server";
+
+    auto addToPoller = [&](zmq::socket_t &socket, zmq::active_poller_t &poller) {
+        poller.add(socket, zmq::event_flags::pollin, [&](zmq::event_flags ef) {
+            zmq::message_t msg;
+            auto result = socket.recv(msg, zmq::recv_flags::dontwait);
+            messageCounter++;
+        });
+    };
+
+    common_server_client_setup sockets(false);
+
+    test_monitor monitor;
+    monitor.init(sockets.server, monitorAddress);
+
+    zmq::active_poller_t poller;
+    monitor.addToPoller(poller);
+    addToPoller(sockets.server, poller);
+
+    sockets.init();
+    sockets.client.send(zmq::message_t(0), zmq::send_flags::dontwait);
+    CHECK(monitor.clientAccepted == 0);
+    CHECK(monitor.clientDisconnected == 0);
+
+    //Act
+    for (int i = 0; i < 10; i++) {
+        poller.wait(std::chrono::milliseconds(10));
```

Comment:
could probably make more robust by increasing the wait time and breaking when monitor.clientAccepted > 0

Reply:
fixed according to your suggestion. In addition applied event filtering to monitor to only look for the desired events and ignore others.